contracts/Canopy.sol

```diff
@@ -161,31 +161,73 @@ contract Canopy {
     
     function rankPosts() internal {}
 
-    // @dev read from mapping
-    function getPostsByTime() external {}
+    // @dev read from mapping - show 100 of the most recent
+    function getMostRecentPost() external {
+        //since IDs are sequential, length of array = most recent post id
+        return posts.length;
+    }
 
     // @dev read from mapping
-    function getScoreById() internal {}
+    function getScoreById(uint _postID) internal {
+        return posts[_postID].score;
+    }
 
     // @dev user can choose when to cash out, make sure to check address
-    function cashOut() external payable onlyPoster {}
+    function cashOut(uint _postId) external payable onlyPoster {
+        //check that payout is to posterAddress with onlyPoster
+        //Setting the Variables required
+        uint _paymentToPoster;
+        uint _bonusPayout;
+        uint _bonusPayoutVoters;
+        uint _totalBonus;
+        uint _totalValue;
+        uint _totalPayout;
+
+        _totalValue = p.valuePositive + p.valueNegative;
+        Post memory p = posts[_postId];
+        _paymentToPoster = p.stake * (p.valuePositive / totalValue) + p.valuePositive;
```

Comment:
we should probably make `ratio` its own variable and define it on a new line using the sqrt value so that this doesn't get gamed.

-> I will add this since the sqrt function is on my branch.